lib/mongo/collection.rb

```diff
@@ -350,10 +350,54 @@ def watch(pipeline = [], options = {})
     # @return [ Integer ] The document count.
     #
     # @since 2.1.0
+    #
+    # @deprecated Use count_documents or estimated_document_count instead.
     def count(filter = nil, options = {})
       View.new(self, filter || {}, options).count(options)
     end
 
+    # Get a count of matching documents in the collection.
+    #
+    # @example Get the number of documents in the collection.
+    #   collection_view.count_documents
+    #
+    # @param [ Hash ] filter A filter for matching documents.
+    # @param [ Hash ] options Options for the count command.
+    #
+    # @option opts :skip [ Integer ] The number of documents to skip.
+    # @option opts :hint [ Hash ] Override default index selection and force
+    #   MongoDB to use a specific index for the query. Requires server version 3.6+.
+    # @option opts :limit [ Integer ] Max number of docs to return.
+    # @option opts :max_time_ms [ Integer ] The maximum amount of time to allow the
+    #   command to run.
+    # @option opts [ Hash ] :read The read preference options.
+    # @option opts [ Hash ] :collation The collation to use.
+    #
+    # @return [ Integer ] The document count.
+    #
+    # @since 2.6.0
+    def count_documents(filter, options = {})
+      View.new(self, filter, options).count_documents(options)
+    end
+
+    # Gets an estimate of the count of documents in a collection using collection metadata.
+    #
+    # @example Get the number of documents in the collection.
+    #   collection_view.estimated_document_count
+    #
+    # @param [ Hash ] options Options for the count command.
```

Comment:
Same here. The count command is an implementation detail here. This helper may use the $collStats aggregate stage in the future.